File: P02 - Kelas Objek/Python/Bag. pertama/Kelas08.py
# File: Kelas08.py
# Penulis : Elmosius Suli
# Accessor & Mutator (getter & setter method)

class Kelas08:
    def __init__(self, age):
       self.__age = age
    
    @property
    def age(self):          # accessor
        return self.__age

    
    @age.setter 
    def age(self, age:int): # mutator
        self.__age = age
        
## Program utama ##
def main():
    k = Kelas08(21)
    # print("Age: ", k.__age) # error karena __age private
    
    print("Age: ", k.age)     # berhasil, lewat accessor / getter
    k.age = 17                # berhasil, lewat mutator / setter
    print("Age: ", k.age)    
    
if __name__ == '__main__':
    main()

File: P02 - Kelas Objek/Java/Bag. pertama/Kelas08.java
package Praktikum.P02;

// Accesor & Mutator (getter & setter method)
public class Kelas08 {
    private int age;

    public Kelas08(int age){   // konstuktor
        this.age = age;
    }

    public int getAge(){      // accessor / getter method
        return this.age;
    }

    public void setAge(int age){    // mutator / setter method
        this.age = age;
    } 

    public static void main(String[] args){
        Kelas08 k;
        k = new Kelas08(21);
        // meskipun age private, masih bisa karena main() berada dalam kelas yg sama

        System.out.println("Age: " + k.age);
        System.out.println("Age: " + k.getAge());   // berhasil, lewat accesor / getter
        k.setAge(17);                           // berhasil, lewat mutator / setter
        System.out.println("Age: " + k.getAge());
    }



}
